File: 485/max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
function findMaxConsecutiveOnes(nums) {
    let len = nums.length;
    if (len == 0)
        return 0;

    let maxConsecutiveOnes = 0;
    for (let i = 0; i < len;) {
        if (nums[i] == 0) {
            i++;
            continue;
        } else {
            let consecutiveOnes = 0;
            while (nums[i] == 1) {
                consecutiveOnes++;
                i++;
            }
            if (consecutiveOnes > maxConsecutiveOnes) {
                maxConsecutiveOnes = consecutiveOnes;
            }
        }
    }
    return maxConsecutiveOnes;
}

File: 485/max-consecutive-ones.go
func findMaxConsecutiveOnes(nums []int) int {
	var maxConsecutiveOnes int
	length := len(nums)
	for i := 0; i < length; {
		if 0 == nums[i] {
			i++
			continue
		} else {
			consecutiveOnes := 0

			for nums[i] == 1 {
				consecutiveOnes++
				i++
			}
			if consecutiveOnes > maxConsecutiveOnes {
				maxConsecutiveOnes = consecutiveOnes
			}
		}
	}
	return maxConsecutiveOnes
}